glow/rate_limiter_test.go

```diff
@@ -0,0 +1,84 @@
+package glow
+
+import (
+	"testing"
+	"time"
+)
+
+func TestRateLimiterSingle(t *testing.T) {
+	rl := NewRateLimiter(1, 50*time.Millisecond)
+
+	ticker := time.NewTicker(20 * time.Millisecond)
+	defer ticker.Stop()
+
+	count := 0
+	allowed := 0
+	for _ = range ticker.C {
+		if rl.Allow() {
+			allowed++
+		}
+		count++
+		if count == 4 {
+			break
+		}
+	}
+
+	t.Logf("allowed %v", allowed)
+
+	if allowed != 2 {
+		t.Errorf("Single test failed, wanted %v got %v", 2, allowed)
+	}
+}
+
+// Basic multithreaded RateLimiter test
+func TestRateLimiterMultithreaded(t *testing.T) {
+
+	maxcalls := 5
+	maxdur := 100 * time.Millisecond
+	testcalls := 19
+	testint := 15 * time.Millisecond
+
+	rl := NewRateLimiter(maxcalls, maxdur)
+
+	ch := make(chan bool, 2*testcalls)
+
+	start := time.Now()
+
+	go simulateCalls(testcalls, testint, rl, ch)
+	go simulateCalls(testcalls, testint, rl, ch)
+
+	var called int
+	var allowed int
+
+	for i := 0; i < 2*testcalls; i++ {
+		if <-ch {
+			called++
+			allowed++
+		} else {
+			called++
+		}
+	}
+
+	t.Logf("allow %v logs in %v, requests %v allowed %v in %v", maxcalls, maxdur, called, allowed, time.Since(start))
```

Comment:
This is another place where the Logf call is just going to create spam in the test suite and make the test suite a lot harder to read.

Reply:
for the most part, depend on `t.Error` and `t.Fatal` to send any messages during testing